File: EmojibleAndroid/app/src/main/java/com/dji/emojibleandroid/models/StudentModel.kt
package com.dji.emojibleandroid.models

class StudentModel(
    email: String,
    name: String,
    surname: String,
    birthDate: String,
    var classId: String?
) : UserModel(email, name, surname, birthDate) {
    constructor(dictionary: HashMap<String, Any>) : this(
        dictionary["email"] as String,
        dictionary["name"] as String,
        dictionary["surname"] as String,
        dictionary["birthDate"] as String,
        dictionary["classId"] as String?
    ) {
        if (classId == "") {
            classId = null
        }
    }

    override var dictionary: HashMap<String, Any> = super.dictionary.let {
        if (classId == null)
            classId = ""
        it["classId"] = classId as Any
        it["userType"] = "Student"
        it
    }
}

File: EmojibleIOS/EmojibleIOS/Models/StudentModel.swift
//
//  StudentModel.swift
//  EmojibleIOS
//
//  Created by Haldun Balim on 12.11.2020.
//

import Foundation
class StudentModel:UserModel{
    var classId: String?
    
    init(email:String, name:String, surname:String, birthDate:Date, classId:String? = nil) {
        self.classId = classId == nil ? "" : classId
        super.init(email:email, name:name, surname:surname, birthDate: birthDate)
    }
    
    //to document
    
    override init(dictionary: [String:Any]) {
        classId = dictionary["classId"] as? String ?? ""
        super.init(dictionary: dictionary)
    }
    
    override var dictionary: [String : Any]{
        var dict = super.dictionary
        dict["classId"] = classId
        dict["userType"] = "Student"
        return dict
    }
}
